rfcs/2fa.md

```diff
@@ -0,0 +1,31 @@
+# 2FA
+The module provides possibility to add second factor to the authentification process.
+
+### Enabling 2FA
+If the user wants to enable 2FA service generates secret key and recovery code.
+The user may authorize secret key in any authentificator app (e.g Google Authentificator).
+From that point user has an ability to generate TOTP (time-based one time password).
+Service verifies that TOTP generated by the user is valid. If it is service attaches
+secret key and recovery code to the user's account.
+
+Recovery code may be used in the future to recover an access to an account in case
+user can't provide a valid TOTP (e.g user lost an access to authentificator app).
+
+### Authentification
+If the user tries to perform some action which requires 2FA service checks that 2FA
+has been enabled by the user. If it was service doesn't perform an action until
+user provides a valid TOTP.
+
+### Disabling 2FA
+To disable 2FA user must provides a valid TOTP. In this case service detaches
+both secret key and recovery code from user's account. From this point user will
+no longer be asked for TOTP during authentification process.
+
+### Actions
+* `generateKey` - generates secret key
+* `regenerateKey` - regenerates secret key (if the user did provide recovery code)
+* `generateRecoveryCode` - generates recovery code which may be used to recover secret key
```

Comment:
not needed either, as this should be a part of `attach`, once we attach -> we return recovery code once